mypy/stubtest.py

```diff
@@ -161,19 +167,20 @@ def test_module(module_name: str) -> Iterator[Error]:
     """
     stub = get_stub(module_name)
     if stub is None:
-        yield Error([module_name], "failed to find stubs", MISSING, None)
+        yield Error([module_name], "failed to find stubs", MISSING, None, runtime_desc="N/A")
         return
 
     try:
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
-            runtime = importlib.import_module(module_name)
-            # Also run the equivalent of `from module import *`
-            # This could have the additional effect of loading not-yet-loaded submodules
-            # mentioned in __all__
-            __import__(module_name, fromlist=["*"])
+        with open(os.devnull, "w") as devnull:
+            with warnings.catch_warnings(), redirect_stdout(devnull), redirect_stderr(devnull):
+                warnings.simplefilter("ignore")
+                runtime = importlib.import_module(module_name)
+                # Also run the equivalent of `from module import *`
+                # This could have the additional effect of loading not-yet-loaded submodules
+                # mentioned in __all__
+                __import__(module_name, fromlist=["*"])
     except Exception as e:
```

Comment:
I made this `BaseException` because I've run into certain modules that call `exit()` or raise a `SystemExit`. If that happens you will only see:
```
> echo "raise SystemExit" > test.py
> stubtest test

>
```

Reply:
Badly behaved modules like that are rare and aren't a good target for stubtest-ing. There are a million pathological things modules could do to mess up stubtest and I'm not looking to play the futile game of guarding against these hypotheticals. stubtest is only going to work on modules that are mostly well behaved.